runtime/sema/check_composite_declaration.go

```diff
@@ -1983,25 +1993,37 @@ func (checker *Checker) checkDefaultDestroyParamExpressionKind(
 		}
 		checker.report(&DefaultDestroyInvalidArgumentError{
 			Range: ast.NewRangeFromPositioned(checker.memoryGauge, arg),
+			Kind:  InvalidIdentifier,
 		})
 
 	case *ast.MemberExpression:
 
+		memberExpressionInfo, ok := checker.Elaboration.MemberExpressionMemberAccessInfo(arg)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		rejectReferenceTypedSubExpressions(memberExpressionInfo.ResultingType)
```

Comment:
In an expression like `self.ref.i` where `i` is of `Int` type, isn't the `ResultingType` also `Int`?

Reply:
Yes, the `ResultingType` of `self.ref.i` is `Int`. However, we will recursively check the `self.ref` subexpression, and that has a `ResultingType` of `&InnerResource`, and will trigger the error.

Reply:
oh nice! Didn't realize it checks the expression recursively 👍